browser/actors/FathomChild.jsm

```diff
@@ -6,12 +6,7 @@
 
 var EXPORTED_SYMBOLS = ["FathomChild"];
 
-// Example for importing a Fathom module:
-// ChromeUtils.defineModuleGetter(
-//   this,
-//   "Fathom",
-//   "resource://gre/modules/Fathom.jsm"
-// );
+const {fathom: {dom, rule, ruleset, score, type, utils: {inlineTextLength}}} = ChromeUtils.import("resource://gre/modules/third_party/fathom/fathom.js");
```

Comment:
TIL: Nested destructuring; interestingly only the most deeply nested values are actually declared -- `fathom` and `utils` are `undefined`.

Reply:
Isn't it cool? It made my brain hurt at first, but I've come to really like it. And the way it integrates with imports, avoiding a special-case syntax, is an actual good piece of language design. Good job, latter-day ES designers!

I agree, the most unobvious piece of the design is that only the leaf nodes get turned into local symbols by `const` or `let`.

Reply:
If you want to really wrap your mind around a pole, try grokking a destructuring function arg with a default value. I think I may even have done that once in the codebase—hopefully with a big fat comment around it.

Reply:
Ah, yes, I did it in `distance()`. It's not anywhere near as hairy as I was imagining, though: I didn't specify defaults for each individual subkey, just defaulted the whole thing to `{}`.